### PR TITLE
[New Docs PR] - Helidon MP Reactive Messaging

### PR DESCRIPTION
**Pull request by danielkec**
Related to #4215

Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

**Review by m0mus — APPROVED** [review]

File: docs/mp/reactivemessaging/kafka.adoc [hunk 1]
```diff
@@ -83,6 +93,69 @@ public PublisherBuilder<String> produceToKafka() {
...
+=== Dead Letter Queue
+
+Sends nacked messages to error topic, link:https://en.wikipedia.org/wiki/Dead_letter_queue[DLQ] is well known pattern for dealing with unprocessed messages.
+
+Helidon can derive connection settings for DLQ topic automatically if the error
+topic is present on the same Kafka cluster.
+Serializers are derived from deserializers used for consumption
+`org.apache.kafka.common.serialization.StringDeserializer` >
+`org.apache.kafka.common.serialization.StringSerializer`.
+In this ideal case only the name of error topic is needed.
```
Comment (ljamen) [inline docs/mp/reactivemessaging/kafka.adoc:117]: Note that the name of the error topic is needed only in this case.

File: docs/mp/reactivemessaging/kafka.adoc [hunk 2]
```diff
@@ -83,6 +93,69 @@ public PublisherBuilder<String> produceToKafka() {
...
+[source,yaml]
+.Example of derived DLQ config:
+----
+mp.messaging:
+  incoming:
+    my-channel:
+      nack-dlq: dql_topic_name
+----
+
+In case custom connection is needed, all the usual producer configuration is possible
```
Comment (ljamen) [inline docs/mp/reactivemessaging/kafka.adoc:128]: If a custom connection is needed, then use the 'nack-dlq' key for all of the producer configuration.

File: docs/mp/reactivemessaging/mock.adoc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+== ToC
+
+- <<Overview, Overview>>
+- <<Emitting data, Emitting data>>
+- <<Asserting data, Asserting data>>
+- <<Configuration, Configuration>>
+- <<Helidon test with Mock Connector, Helidon Test>>
+
+== Overview
+Mock connector is a simple application scoped bean that can be used for emitting to a channel
```
Comment (ljamen) [inline docs/mp/reactivemessaging/mock.adoc:37]: application-scoped

File: docs/mp/reactivemessaging/mock.adoc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.messaging.mock</groupId>
+    <artifactId>helidon-messaging-mock</artifactId>
+</dependency>
+----
+
+WARNING: Mock connector should be used in the test environment only!
```
Comment (ljamen) [inline docs/mp/reactivemessaging/mock.adoc:51]: Caution: should be used here. We restrict warnings for bodily harm. 

Caution: The mock connector should only be used in test environments.

File: docs/mp/reactivemessaging/mock.adoc [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+|Key            |Default value   | Description
+|mock-data      |                | Initial data emitted to the channel immediately after subscription
+|mock-data-type |java.lang.String| Type of the emitted initial data to be emitted
+|===
+
+== Helidon test with Mock Connector
+Mock connector works great with built-in Helidon test support for
+link:/{rootdir}/testing/testing.adoc[JUnit 5] or link:/{rootdir}/testing/testing-ng.adoc[Test NG].
+
+As Helidon test support makes a bean out of your test, you can inject MockConnector right in to it.
```
Comment (ljamen) [inline docs/mp/reactivemessaging/mock.adoc:95]: Mock Connector directly into it or is it MockConnector when referring to the bean?

File: docs/se/reactivemessaging/introduction.adoc [hunk 6]
```diff
@@ -121,7 +121,7 @@ implicit ack before callback is executed is necessary.
 
 === Connector
 Connector concept is a way for connecting <<Channel,channels>> to external sources.
```
Comment (ljamen) [inline docs/se/reactivemessaging/introduction.adoc:123]: Connector concept? Does this mean the concept of connecting channels to external sources or could it be "Connectors are used to connect channels to external sources."?

File: docs/se/reactivemessaging/introduction.adoc [hunk 7]
```diff
@@ -121,7 +121,7 @@ implicit ack before callback is executed is necessary.
...
-To make xref:connector.adoc[creation and usage of connectors]
+To make creation and usage of connectors
```
Comment (ljamen) [inline docs/se/reactivemessaging/introduction.adoc:124]: To simplify the creation and usage of connectors, Helidon SE Messaging...

File: docs/se/reactivemessaging/introduction.adoc [hunk 8]
```diff
@@ -121,7 +121,7 @@ implicit ack before callback is executed is necessary.
...
 as easy and versatile as possible, Helidon SE Messaging uses same API for connectors
 like xref:../../mp/reactivemessaging/introduction.adoc[MicroProfile Reactive Messaging] does.
 This allows connectors to be usable in both flavors of Helidon with one limitation which is
```
Comment (ljamen) [inline docs/se/reactivemessaging/introduction.adoc:127]: to be used in both Heldion MP and SE, however it has to be able to work without CDI.

**Review by ljamen — COMMENTED** [review]
Comments inline. A couple of questions re: meaning. We will check the headings once we have established the style guide. I did not mark those now.

**Comment by ljamen** [comment]
@danielkec Hi - I can make these updates in another PR and have you review them. Is that ok? I don't want to conflict with your changes if you've already started?

**Pull request merged**